contracts/staking/TribalChiefSyncV2.sol

```diff
@@ -0,0 +1,191 @@
+pragma solidity ^0.8.0;
+
+import "./ITribalChief.sol";
+
+interface IAutoRewardsDistributor {
+    function setAutoRewardsDistribution() external;
+}
+
+interface ITimelock {
+    function execute(
+        address target,
+        uint256 value,
+        bytes calldata data,
+        bytes32 predecessor,
+        bytes32 salt
+    ) external;
+}
+
+/**
+    @title TribalChief Synchronize contract
+    This contract is able to keep the tribalChief and autoRewardsDistributor in sync when either:
+    1. adding pools or 
+    2. updating block rewards
+
+    It needs the EXECUTOR role on the optimistic timelock, so it can atomically trigger the 3 actions.
+
+    It also includes a mapping for updating block rewards according to the schedule in https://tribe.fei.money/t/tribe-liquidity-mining-emission-schedule/3549
+    It needs the TRIBAL_CHIEF_ADMIN_ROLE role to auto trigger reward decreases.
+ */
+contract TribalChiefSyncV2 {
+    ITribalChief public immutable tribalChief;
+    IAutoRewardsDistributor public immutable autoRewardsDistributor;
+    ITimelock public immutable timelock;
+
+    /// @notice a mapping from reward rates to timestamps after which they become active
+    mapping(uint256 => uint256) public rewardsSchedule;
+
+    uint256[] internal rewardsArray;
```

Comment:
prefer making this public so it's easy to pull all values out of the rewardsSchedule mapping